server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -519,9 +776,16 @@ public void setDependentProductIds(Set<String> dependentProductIds) {
     @Override
     public String getHref() {
         // If we don't have an owner here, we're in a bit of trouble.
-        return (this.getOwner() != null && this.getOwner().getKey() != null && this.getId() != null) ?
-            "/owners/" + this.getOwner().getKey() + "/products/" + this.getId() :
-            "";
+        // PER-ORG PRODUCT VERSIONING TODO:
+        // Do we have a way to determine the current owner from context? We need a way to reference
+        // a specific product (uuid) but with owner permissions...? Should this just change to UUIDs
+        // everywhere? This is messy.
+
```

Comment:
IMO getHref() should point to this exact product -- using the uuid as you do below -- and should not be based on the context of the current Owner. Permissions on GET /product/:uuid  would then have to be modified to check the Principal's Owner in order to view it via the ProductResource.